src/firmware/app/logger/logger.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include "shared/proto/robot_log_msg.nanopb.h"
+
+/**
+ * Initializes the logger with the given robot id and the log handler function
+ *
+ * @param robot_id The robot id the logs are coming from
+ * @param robot_log_msg_handler The function to handle RobotLog protos
+ */
+void app_logger_init(unsigned robot_id,
+                     void (*robot_log_msg_handler)(TbotsProto_RobotLog log_msg));
+
+/**
+ * Creates a new RobotLog proto and calls the the robot_log_msg_handler
```

Comment:
Could you explain how the logging process will work? Do we call app_logger_init for each robot and if we want to log something for that particular robot, we call either TLOG_DEBUG, TLOG_INFO, etc.?
Why do lines 27-34, have the hard-coded file and line values, is every log that is going to logged in the same file and line number? 
 (Also there are two "the"s in line 14)

Reply:
> Could you explain how the logging process will work? Do we call app_logger_init for each robot and if we want to log something for that particular robot, we call either TLOG_DEBUG, TLOG_INFO, etc.?

Yup thats it, `app_logger_init` will already be called in the main function before any tasks start running, so we just use the macros to log

> Why do lines 27-34, have the hard-coded file and line values, is every log that is going to logged in the same file and line number?

`__LINE__` and `__FILE__` are predefined macros that the pre-processor will replace with the line number and file name. So the log will contain info about the line number and file name of where its being used. 
Added a comment to clarify

>  (Also there are two "the"s in line 14)
Fixed!